internal/hcp/internal/controllers/link/controller.go

```diff
@@ -114,39 +114,30 @@ func (i *linkInitializer) Initialize(ctx context.Context, rt controller.Runtime)
 	}
 
 	// Create the link resource for a configuration-based link
-	w := retry.Waiter{
-		MinWait: 1 * time.Second,
-		MaxWait: 15 * time.Second,
-		Jitter:  retry.NewJitter(50),
-	}
-	retryCtx, cancel := context.WithTimeout(ctx, 1*time.Minute)
-	defer cancel()
-	for {
-		_, writeErr := rt.Client.Write(ctx,
-			&pbresource.WriteRequest{
-				Resource: &pbresource.Resource{
-					Id: &pbresource.ID{
-						Name: types.LinkName,
-						Type: pbhcp.LinkType,
-					},
-					Metadata: map[string]string{
-						MetadataSourceKey: MetadataSourceConfig,
-					},
-					Data: data,
+	_, err = rt.Client.Write(ctx,
+		&pbresource.WriteRequest{
+			Resource: &pbresource.Resource{
+				Id: &pbresource.ID{
+					Name: types.LinkName,
+					Type: pbhcp.LinkType,
+				},
+				Metadata: map[string]string{
+					MetadataSourceKey: MetadataSourceConfig,
 				},
+				Data: data,
 			},
-		)
-		if writeErr != nil {
-			// Retry on error
-			rt.Logger.Trace("error initializing link, retrying", "error", writeErr)
-			if err := w.Wait(retryCtx); err != nil {
-				// Retry timed out, return last write error
-				return writeErr
-			}
-			continue
+		},
+	)
+	if err != nil {
+		if strings.Contains(err.Error(), storage.ErrWrongUid.Error()) ||
+			strings.Contains(err.Error(), "leader unknown") {
```

Comment:
Are these underlying errors wrapped? Can we use `errors.Is()` instead?

Reply:
No, they're gRPC status errors, so I couldn't get `errors.Is()` working since the error has been converted to a string to set as the `Message`. Though rather than `string.Contains`, I could probably check for the exact string match of the error message.

Reply:
Oh but I guess checking the exact string has the downside of not being able to deal with changes to the errors, especially if they get wrapped again. 🤔 I could do something like create an error from the status message and then check `errors.Is()`, but that seems like it would add more complexity than benefit.

Reply:
I'd like to leave this as is for now, but let me know if you'd rather me pursue more explicit error checking!

Reply:
I'm ok to leave this for now, but in the future I hope we can investigate a nicer way to do this through the gRPC api using concrete errors.